core/src/main/java/org/apache/iceberg/rest/RESTSerializers.java

```diff
@@ -110,7 +110,7 @@ public static class TableMetadataSerializer extends JsonSerializer<TableMetadata
     @Override
     public void serialize(TableMetadata metadata, JsonGenerator gen, SerializerProvider serializers)
         throws IOException {
-      TableMetadataParser.toJson(metadata, gen);
+      TableMetadataParser.toJson(metadata, null, gen);
```

Comment:
Wouldn't providing null here for FileIO cause IllegalArgumentException in BaseSnapshot.cacheManifests(FileIO)?

https://github.com/apache/iceberg/blob/c0f175b517edef5dd1b06a5ac7890fb35224bc25/core/src/main/java/org/apache/iceberg/BaseSnapshot.java#L178

Reply:
this might indeed cause issues, especially in https://github.com/apache/iceberg/blob/14d567431206afcecbf02e9351f6b08033b55dfe/core/src/main/java/org/apache/iceberg/SnapshotParser.java#L170-L177 with `write.manifest-lists.enabled=false` and with a `null` FileIO

Reply:
I have revisited the code around this area and having a dependency on `FileIO` in Snapshot for the single purpose of reading/writing `manifests` turned out to be quite tricky. However, I think I found an acceptable approach to deal with this in https://github.com/apache/iceberg/pull/5734.